### DISCUSSION TITLE
How to use backrefs with stateful lexer?

### DISCUSSION THREAD

**Discussion by ssquirrel** [alecthomas/participle#272] [Q&A] [answered]
I would like to parse blocks starting and ending with the same string, eg.
```
\start XXX
    ...
\end XXX
```
The XXX should match. And I do want to capture XXX once. It's the name of the block.

**Comment by alecthomas** [comment, marked as answer]
Use `\1` in a pushed state to refer to the capture groups from the rule that did the push. There's an example in [the tests](https://github.com/alecthomas/participle/blob/master/lexer/stateful_test.go#L70-L83).

**Reply by alecthomas** [reply]
This is not the right use case for the stateful lexer though. I would just capture the identifiers in the parser, and verify that that match later during an analysis pass.

**Reply by ssquirrel** [reply]
Thanks a lot!
